### ISSUE TITLE
AwsDefaultClientBuilder still tries to access ProfileFile even when supplied with StaticCredentialsProvider 

### ISSUE THREAD

**Issue by rounakraj8** [aws/aws-sdk-java-v2#3269]
### Describe the bug

I need to use  `AwsBasicCredentials` when creating a `FirehoseAsyncClient`, so I use `StaticCredentialsProvider.create()` as the credential provider, but the build method internally calls a method that requires ProfileFile (~/.aws/credentials). I think if I am passing AwsBasicCredentials using StaticCredentialsProvider, SDK should not try accessing profile/ProfileFile.

       AwsBasicCredentials credentials = AwsBasicCredentials.create(awsAccessKey, awsSecretKey);

        _client = FirehoseAsyncClient.builder()
                .credentialsProvider(StaticCredentialsProvider.create(credentials))
                .region(Region.of(firehoseRegion))
                .overrideConfiguration(clientOverrideConfiguration)
                .build();
                
PS: I am running this code inside a sandbox environment which doesn't have permission to read  `~/.aws/credentials`. This is causing exception and my program execution stops.               ``

### Expected Behavior

SDK should not try to access `~/.aws/credentials` as credentials are passed via  StaticCredentialsProvider.

### Current Behavior

The sandbox environment is throwing `java.io.FilePermission`

### Reproduction Steps

Try to create a client with `AwsBasicCredentials`, and use `StaticCredentialsProvider`. Call trace shows that it's trying to read ProfileFile('~/.aws/credentials').

### AWS Java SDK version used

2.17.217

### JDK version used

1.8

### Operating System and version

Linux

**Comment by debora-ito** [comment]
@rounakraj8 can you show the full stacktrace with the error?

**Comment by rounakraj8** [comment]
Here's the stacktrace, the AWS SDK is used in a Java app that serves as a Java callout from Apigee.

`java.security.AccessControlException: access denied ("java.io.FilePermission" "/opt/apigee/.aws/credentials" "read") at java.security.AccessControlContext.checkPermission(AccessControlContext.java:472) at java.security.AccessController.checkPermission(AccessController.java:884) at java.lang.SecurityManager.checkPermission(SecurityManager.java:549) at com.apigee.securitypolicy.InternalSecurityManager.checkPermission(InternalSecurityManager.java:85) at java.lang.SecurityManager.checkRead(SecurityManager.java:888) at sun.nio.fs.UnixPath.checkRead(UnixPath.java:795) at sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:49) at sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:144) at sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99) at java.nio.file.Files.readAttributes(Files.java:1737) at java.nio.file.Files.isRegularFile(Files.java:2229) at software.amazon.awssdk.profiles.ProfileFileLocation.lambda$resolveIfExists$2(ProfileFileLocation.java:90) at java.util.Optional.filter(Optional.java:178) at software.amazon.awssdk.profiles.ProfileFileLocation.resolveIfExists(ProfileFileLocation.java:90) at software.amazon.awssdk.profiles.ProfileFileLocation.credentialsFileLocation(ProfileFileLocation.java:77) at software.amazon.awssdk.profiles.ProfileFile.addCredentialsFile(ProfileFile.java:138) at software.amazon.awssdk.utils.builder.SdkBuilder.applyMutation(SdkBuilder.java:61) at software.amazon.awssdk.profiles.ProfileFile.defaultProfileFile(ProfileFile.java:90) at software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder.mergeGlobalDefaults(SdkDefaultClientBuilder.java:258) at software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder.asyncClientConfiguration(SdkDefaultClientBuilder.java:201) at software.amazon.awssdk.services.firehose.DefaultFirehoseAsyncClientBuilder.buildClient(DefaultFirehoseAsyncClientBuilder.java:28) at software.amazon.awssdk.services.firehose.DefaultFirehoseAsyncClientBuilder.buildClient(DefaultFirehoseAsyncClientBuilder.java:22) at software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder.build(SdkDefaultClientBuilder.java:145) at com.sophos.central.apigee.callout.logger.Logger.setupFirehoseClient(Logger.java:127) at com.sophos.central.apigee.callout.logger.Logger.initialize(Logger.java:102) at com.sophos.central.apigee.callout.logger.Logger.execute(Logger.java:70) at com.apigee.steps.javacallout.JavaCalloutStepDefinition$ClassLoadWrappedExecution.execute(JavaCalloutStepDefinition.java:235) at com.apigee.steps.javacallout.JavaCalloutStepDefinition$SecurityWrappedExecution$1.run(JavaCalloutStepDefinition.java:302) at com.apigee.steps.javacallout.JavaCalloutStepDefinition$SecurityWrappedExecution$1.run(JavaCalloutStepDefinition.java:300) at java.security.AccessController.doPrivileged(Native Method) at com.apigee.steps.javacallout.JavaCalloutStepDefinition$SecurityWrappedExecution.execute(JavaCalloutStepDefinition.java:300) at com.apigee.steps.javacallout.JavaCalloutStepDefinition$CallOutWrapper.execute(JavaCalloutStepDefinition.java:169) at com.apigee.messaging.runtime.steps.StepExecution.execute(StepExecution.java:157) at com.apigee.flow.execution.AbstractAsyncExecutionStrategy$AsyncExecutionTask.call(AbstractAsyncExecutionStrategy.java:83) at com.apigee.flow.execution.AbstractAsyncExecutionStrategy$AsyncExecutionTask.call(AbstractAsyncExecutionStrategy.java:48) at com.apigee.threadpool.CallableWrapperForMDCPreservation.call(CallableWrapperForMDCPreservation.java:26) at com.apigee.threadpool.ThreadPoolManager$QueueAwareCallableTask.call(ThreadPoolManager.java:546) at java.util.concurrent.FutureTask.run(FutureTask.java:266) at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) at java.util.concurrent.FutureTask.run(FutureTask.java:266) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) at java.lang.Thread.run(Thread.java:748)`

**Comment by rounakraj8** [comment]
@debora-ito I have added the stacktrace, please have a look.

**Comment by debora-ito** [comment]
@rounakraj8 I see you also have a Support Case open with some answers for your questions. I'm marking this to auto-close, let us know if you have any other question.

**Comment by csteger81590** [comment]
@rounakraj8  I am currently having the same problem with the SDK v2 and I was wondering if you ever found a resolution?

**Comment by rounakraj-sophos** [comment]
@csteger81590 I got a reply from AWS support. It suggested using ClientOverrideConfiguration.Builder.profileFile(...).

We tried using ClientOverrideConfiguration.Builder.profileFile, but now the issue is that AWS Java SDK 2 has a dependency on a Netty library, and this library uses reflection. The sandbox on which my code was running doesn't allow reflection and throws an exception(AccessControlException).

Maybe your sandbox doesn't have a similar restriction and it will work for you.

**Comment by rounakraj8** [comment]
@csteger81590 I got a reply from AWS support. It suggested using ClientOverrideConfiguration.Builder.profileFile(...).

I tried using ClientOverrideConfiguration.Builder.profileFile, but now the issue is that AWS Java SDK 2 has a dependency on a Netty library, and this library uses reflection. The sandbox on which my code was running doesn't allow reflection and throws an exception(AccessControlException).

Maybe your sandbox doesn't have a similar restriction and it will work for you.
